### PR TITLE
Installer to bundle

### PR DESCRIPTION
**Pull request by craigh**
Rewrite of installer to make it 2.0.0 compatible

| Q                 | A
| ----------------- | ---
| Bug fix?          | yes
| New feature?      | yes
| BC breaks?        | yes - removed SQL installer
| Deprecations?     | no
| Tests pass?       | yes
| Fixed tickets     | see list below
| Refs tickets      | see list below
| License           | MIT
| Doc PR            | included
| Changelog updated | YES

 - [x] Conversion of Install and Upgrade routines to controller-based bundle
 - [x] Use of symfony forms and twig
 - [x] Wizard class to assist with multi-stage user interaction
 - [x] CLI-based  **two** step installer
 - [x] CLI-based upgrader
 - [x] implement [Forms on the CLI](https://github.com/matthiasnoback/symfony-console-form) (this lib requires php >= 5.4 because it uses Traits.)
 - [x] Removed sql-installer
 - [x] Update documentation mucho
 - [x] Added entities to Settings and Categories module to allow for installation of legacy tables instead of using DBUtil.
 - [x] auto-creates `custom_parameters.yml` and `personal_config.php` files instead of using `parameters.tml` and `config.php` so distribution files are never edited.
 - [x] A bunch of stuff I can't remember.

Known Issues:
-------------------
 - ***THIS IS FIXED*** (or at least worked around - the bug still exists in Doctrine, but shouldn't be a problem for us) [Doctrine bug 1057](http://www.doctrine-project.org/jira/browse/DBAL-1057) which requires DBAL to ping the DB to get version information. **workaround:** add `server_version` param to `config.yml`
```yaml
doctrine:
    dbal:
        connections:
            default:
                ...
                server_version: 5.5
```
 - CSRF protection on installer forms is ***disabled***.
 - Post-install and post-upgrade admin login is not working.

Tickets Addressed:
------------------------
refs: #2154, #2171, #2162, #2161, #2158, #2144, #2167

### CODE REVIEW

**Comment by craigh** [comment]
I need this to be tested by others. ping @Guite @cmfcmf @drak @shefik @espaan @Kaik @Alexalter @rallek @Portugao @phaidon @rojblake

To clone this to your local test environment:
```Shell
git checkout -b craigh-InstallerToBundle 1.4
git pull https://github.com/craigh/core.git InstallerToBundle
```
You **will need to employ the workaround** for the Doctrine bug mentioned in the PR description in order for this to work. 

tests to perform:
 - install from web interface
 - install from CLI
 - upgrade (from 1.3.5+) from web interface
 - upgrade (from 1.3.5+) from CLI

your help is greatly appreciated.

**Comment by rallek** [comment]
sorry can't test. I have no idea how to proceed. I have no composer/Webserver available on my windows machine. I always take the ci.zikula.org build for tests.

**Comment by craigh** [comment]
here is a (250+ MB) package to download and try if you don't/can't do git. https://drive.google.com/file/d/0B1HnwOnwDnaESXgwdDd0N1FROW8/view?usp=sharing

File: src/lib/Zikula/Bundle/CoreInstallerBundle/Command/InstallCommand.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            ->setName('zikula:install');
+    }
+
+    protected function execute(InputInterface $xinput, OutputInterface $output)
+    {
+        define('_ZINSTALLVER', \Zikula_Core::VERSION_NUM);
+        $commandNames = array(
+            0 => "zikula:install:getdata",
+            1 => "zikula:install:createtables",
+            2 => "zikula:install:finish",
```
Comment (cmfcmf) [inline src/lib/Zikula/Bundle/CoreInstallerBundle/Command/InstallCommand.php]: Is this command outdated? The commands above don't exist.

Comment (craigh) [inline src/lib/Zikula/Bundle/CoreInstallerBundle/Command/InstallCommand.php]: yes. was only temporarily left in the PR hoping that I could figure out a way to make the whole command work, but that seems less and less likely. In order to do so, there would need to be a way to reload the entire container, but from what I can see this is impossible.

**Comment by Kaik** [comment]
Hi, I have done some tests and web installation went smoothly + very nice installer. 
CLI install went all ok apart that this showed up when I checked site (this is first attempt)
no title

Upgrade I need to play more with this to see what is happening looks like on web is redirecting me wrong to localhost so I cannot go further than user login screen... (this is probably server miss-configuration on my side)
console is hmm

**Comment by craigh** [comment]
> CLI install went all ok apart that this showed up when I checked site (this is first attempt)

do you know why it is looking for `config_.yml` (note the underscore)

**Comment by craigh** [comment]
I've just learned/figured out that the [Forms on the CLI lib](https://github.com/matthiasnoback/symfony-console-form) requires `php>=5.4` because it uses Traits. Should this effort be abandoned for now then? ping @cmfcmf @Guite refs: #2161

**Comment by Kaik** [comment]
I have no idea :) why it is looking for config_.yml Looks like second part of file (after underscore) is missing.

web install works still nice
web upgrade : Unable to generate a URL for the named route "zikulaadminmodule_admin_adminpanel" as such route does not exist.

cli install: config yml 
cli upgrade:
*** UPGRADING TO ZIKULA CORE v1.4.0 ***
Please enter the username of the new Zikula admin user default:[Kaik]: 
Please enter the password of the new Zikula admin user default:[xxxxx]: 
Please enter the host where you install Zikula, e.g. "example.com". Do not include subdirectories: ml.dev.com
Please enter the scheme of where you install Zikula, can be either "http" or "https" default:[http]: 
Please enter the url path of the directory where you install Zikula, leave empty if you install it at the top level. Example: /my/sub-dir default:[]: 
Login
Logged in as admin
Upgrade modules

[Symfony\Component\Debug\Exception\ContextErrorException]

Notice: Undefined index: database_name

**Comment by rallek** [comment]
I am trying your download src.zip
it is unzipped on my account
I created a new subdomain for the tryout. 
app/logs and app/cache is writable, 
config.yml has added the server_version: 5.5.  (is 5.5 alsways the right decission?)

But after calling my new subdomain I am directed to http://localhost/install

**Comment by craigh** [comment]
server_version is whatever your version is :-) not necessarily 5.5 (mine is 5.5 locally)

**Comment by Kaik** [comment]
>But after calling my new subdomain I am directed to http://localhost/install
well this is similar what is happening to me on upgrade

**Comment by craigh** [comment]
@Kaik - it sounds like you didn't follow the upgrade instructions (read the full file) but here is the point I think you missed:

> Copy your new `config/config.php` to `config/personal_config.php` and update `config/personal_config.php` with your database settings.

**Comment by rallek** [comment]
Is server_version the php version?

**Comment by craigh** [comment]
> Is server_version the php version?

no the mysql version

**Comment by rallek** [comment]
Ok. And I made the same mistake than Kaik I guess. Will try tomorrow again.

**Comment by craigh** [comment]
> I have no idea :) why it is looking for config_.yml Looks like second part of file (after underscore) is missing.

@Kaik - this means your `env` parameter is missing in `personal_config.yml`

**Comment by Kaik** [comment]
Hmm the config_ issue is on install so it should all be auto generated I will check that as soon I will finish with upgrade part. I have not followed upgrade docs :) now 
---------------------------
| Zikula Upgrader Script |
---------------------------
*** UPGRADING TO ZIKULA CORE v1.4.0 ***
Please enter the username of the new Zikula admin user default:[admin]: Kaik
Please enter the password of the new Zikula admin user: 
Please enter the host where you install Zikula, e.g. "example.com". Do not include subdirectories: core.dev.com
Please enter the scheme of where you install Zikula, can be either "http" or "https" default:[http]: 
Please enter the url path of the directory where you install Zikula, leave empty if you install it at the top level. Example: /my/sub-dir default:[]: 
Login
Logged in as admin
Upgrade modules
Modules upgraded
Install Zikula Routes Module
[ReflectionException]                     
  Class Dizkus_HookHandlers does not exist  

Now let me delete those entries it should pop out once more on scribite hook handlers

**Comment by craigh** [comment]
there should be no third party modules present during upgrade. docs say: **delete all files**

**Comment by Kaik** [comment]
There's nothing there apart from breakeit, profile and legal this is from db

**Comment by craigh** [comment]
> There's nothing there apart from breakeit, profile and legal this is from db

ah. dang it. let me know what happens.

**Comment by craigh** [comment]
you can try running the upgrade command like this:
```Shell
% php app/console zikula:upgrade -vvv
```

this will provide you a trace of the error. please share that.

**Comment by Kaik** [comment]
I was just thinking about that the output is far too short forget about that verbose option

**Comment by Kaik** [comment]
There we have Scribite
```                                              
  [ReflectionException]                       
  Class Scribite_HookHandlers does not exist  
                                              


Exception trace:
 () at /home/Kaik/public_html/core/src/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:962
 ReflectionClass->__construct() at /home/Kaik/public_html/core/src/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:962
 Symfony\Component\DependencyInjection\ContainerBuilder->createService() at /home/Kaik/public_html/core/src/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:490
 Symfony\Component\DependencyInjection\ContainerBuilder->get() at /home/Kaik/public_html/core/src/lib/legacy/Zikula/ServiceManager.php:117
 Zikula_ServiceManager->get() at /home/Kaik/public_html/core/src/lib/legacy/Zikula/HookDispatcher.php:310
 Zikula_HookDispatcher->loadRuntimeHandlers() at /home/Kaik/public_html/core/src/lib/legacy/Zikula/HookDispatcher.php:348
 Zikula_HookDispatcher->reload() at /home/Kaik/public_html/core/src/lib/legacy/Zikula/HookDispatcher.php:117
 Zikula_HookDispatcher->registerSubscriberBundle() at /home/Kaik/public_html/core/src/lib/util/HookUtil.php:155
 HookUtil::registerSubscriberBundles() at /home/Kaik/public_html/core/src/system/ZikulaRoutesModule/Base/RoutesModuleInstaller.php:67
 Zikula\RoutesModule\Base\RoutesModuleInstaller->install() at /home/Kaik/public_html/core/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxInstallController.php:160
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxInstallController->installModule() at /home/Kaik/public_html/core/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:88
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->installRoutesModule() at /home/Kaik/public_html/core/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:70
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->executeStage() at /home/Kaik/public_html/core/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:54
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->commandLineAction() at /home/Kaik/public_html/core/src/lib/Zikula/Bundle/CoreInstallerBundle/Command/UpgradeCommand.php:108
 Zikula\Bundle\CoreInstallerBundle\Command\UpgradeCommand->execute() at /home/Kaik/public_html/core/src/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php:253
 Symfony\Component\Console\Command\Command->run() at /home/Kaik/public_html/core/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:882
 Symfony\Component\Console\Application->doRunCommand() at /home/Kaik/public_html/core/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:195
 Symfony\Component\Console\Application->doRun() at /home/Kaik/public_html/core/src/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:96
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /home/Kaik/public_html/core/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at /home/Kaik/public_html/core/src/app/console:18
```
I will delete all of the hooks see what happens next :)

**Comment by craigh** [comment]
perhaps if the module was present then the error wouldn't occur?

**Comment by Kaik** [comment]
This is what happened 
```
Modules upgraded
Install Zikula Routes Module


                                                             
  [Symfony\Component\Debug\Exception\ContextErrorException]  
  Notice: Undefined offset: 0                                
                                                             


Exception trace:
 () at /home/Kaik/public_html/core/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:115
 Symfony\Component\Debug\ErrorHandler->handleError() at /home/Kaik/public_html/core/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:115
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->installRoutesModule() at /home/Kaik/public_html/core/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:70
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->executeStage() at /home/Kaik/public_html/core/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:54
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->commandLineAction() at /home/Kaik/public_html/core/src/lib/Zikula/Bundle/CoreInstallerBundle/Command/UpgradeCommand.php:108
 Zikula\Bundle\CoreInstallerBundle\Command\UpgradeCommand->execute() at /home/Kaik/public_html/core/src/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php:253
 Symfony\Component\Console\Command\Command->run() at /home/Kaik/public_html/core/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:882
 Symfony\Component\Console\Application->doRunCommand() at /home/Kaik/public_html/core/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:195
 Symfony\Component\Console\Application->doRun() at /home/Kaik/public_html/core/src/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:96
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /home/Kaik/public_html/core/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at /home/Kaik/public_html/core/src/app/console:18
```

**Comment by craigh** [comment]
if you just reran the upgrade, I think it will fail. you have to back all the way up and start over

**Comment by Kaik** [comment]
This is what I am doing remove tables -> import > empty hook tables -> clear cache -> upgrade

**Comment by craigh** [comment]
> This is what I am doing remove tables -> import > empty hook tables -> clear cache -> upgrade

if it works terrific, but this is not a viable plan for the general populace. @Guite any ideas on how to fix this?

**Comment by rallek** [comment]
I am on a new installation (no upgrade). My server version is 5.6 (changed the config.yml). But I am still directed to localhost. I am doing a webinstall.

**Comment by rallek** [comment]
Further analysing: The web install is not asking for anything. A personal_config.php is not there nor personal_config.yml. The INSTALL-1.4.0.md is not telling anything about creating something like this.

PHP is 5.4.34
MySQL is 5.6.31

cmfcmf commits are not included.

Is there anything else I do have to check?

**Comment by craigh** [comment]
>  The web install is not asking for anything.

I'm not entirely following you. The web installer doesn't ask for anything - you set all the parameters by filling out the forms and it creates the `custom_parameters.yml` and `config/personal_config.php` files for you.

**Comment by rallek** [comment]
Which forms?

**Comment by Kaik** [comment]
@rallek there are forms on web installation where you type in all db data / admin data like in every zikula installation. Might be you had these two custom_parameters.yml and config/personal_config.php already filled and installation skipped those steps as data was already in or cache...
I have tried on plain localhost no sub-domain/subfolder so it redirects me to localhost anyway and it worked.

**Comment by rallek** [comment]
Ok. This forms I have never seen.

**Comment by espaan** [comment]
Hi, I don't have time to dive into this at the moment :-(
I like the idea of being able to install from CommandLine, but in the end all ways are workable for me.

**Comment by Kaik** [comment]
@rallek so it look's like it is redirecting you straight to localhost before displaying forms (is it displaying anything like first screen with big install button?) while on upgrade I have seen admin login form and then it redirects me to localhost (that's on subdomain). On localhost* web installation is working nice of course because it redirects to right place.

By localhost I mean there is no subdomain no subdirectory in url only localhost/install or localhost/upgrade

**Comment by craigh** [comment]
this is the code for the redirect:
```php
            $this->container->get('router')->getContext()->setBaseUrl($request->getBasePath()); // compensate for sub-directory installs
            $url = $this->container->get('router')->generate('install', array(), true);
            $response = new RedirectResponse($url);
            $response->send();
```

@cmfcmf any idea why this could fail?

**Comment by Kaik** [comment]
So $request->getBasePath() should be our subdomain etc? but looks like for me and @rallek it is localhost.

**Comment by craigh** [comment]
`$request->getBasePath()` should return anything after the original url. For example, if zikula is installed at `http://www.example.com/core/` then the `basepath` should resolve to `/core` but the basepath isn't the problem here, it is the routing of the main `url` (e.g. `http://www.example.com`). This is returning `localhost` in @rallek 's case instead of the actual domain. I'm hoping @cmfcmf will have the answer because he is a genius. :boat:

**Comment by cmfcmf** [comment]
As far as I understand, @rallek's problem is that it redirects him from `http://test.localhost` to `http://localhost/install` whereas is should be `http://test.localhost/install`. @rallek right?

**Comment by craigh** [comment]
nope.

it redirects from `http://my.domain.com/` to `http://localhost/install`

**Comment by cmfcmf** [comment]
> nope.
>
it redirects from `http://my.domain.com/` to `http://localhost/install`

Well, it might or might not be related to wrong settings in `parameters.yml` or `custom_parameters.yml`. @rallek could you please post both (without passwords of course :smile: )

**Comment by craigh** [comment]
on a standard install, the `parameters.yml` should be the one that ships with zikula. the other shouldn't exist (yet). that may be the problem - that the shipping version of `parameters.yml` has `localhost` ?

**Comment by craigh** [comment]
> @rallek could you please post both

I checked this for him and there was no `custom_parameters.yml` yet (which is right) and the `parameters.yml` was just the stock version from the repo with no changes (which is also right).

**Comment by craigh** [comment]
Travis is failing now because I added the lib that requires php >=5.4.0 but the code itself compensates for the use of either.

**Comment by craigh** [comment]
adding your Database server_version is no longer required! :birthday: :tada:

**Comment by craigh** [comment]
@rallek - I've confirmed the problem you are having, but no solution yet.

**Comment by craigh** [comment]
@rallek - I'm pretty sure the problem is fixed.

**Comment by craigh** [comment]
CI BUILD AVAILABLE
=================

http://ci.zikula.org/job/ZikulaInstallerRefactor/

**Comment by craigh** [comment]
@Kaik give it another shot. the redirect to localhost is fixed. not sure about the Hook thing though.

**Comment by Kaik** [comment]
@craigh ok I will do all the testing today/tonight and will let you know.

**Comment by Portugao** [comment]
Try to check on weekend.

**Comment by craigh** [comment]
@Kaik I am hoping that https://github.com/craigh/core/commit/ff20010fcbbd54b3d7216141b0d3cb448c323154 will alleviate the errors you were experiencing. You should try to run the upgrade without emptying any hook_* tables please.

**Comment by Kaik** [comment]
Ok some results:
Install via Web Interface works great on localhost as well as on domain.
Install via Cli - everything is ok in console no errors etc. checking homepage gives me this error (both config files are present) 
localhost

**Comment by craigh** [comment]
try force-reload in browser a couple times. if that does nothing, try manual delete of app/cache/`<env>`

**Comment by Kaik** [comment]
Upgrade via cli gives me 
'''

Unable to generate a URL for the named route "zikulaadminmodule_admin_adminpanel" as such route does not exist.

500 Internal Server Error - RouteNotFoundException

Stack Trace

in app/cache/dev/appDevUrlGenerator.php at line 53   -
    public function generate($name, $parameters = array(), $referenceType = self::ABSOLUTE_PATH)
    {
        if (!isset(self::$declaredRoutes[$name])) {
            throw new RouteNotFoundException(sprintf('Unable to generate a URL for the named route "%s" as such route does not exist.', $name));
        }
        list($variables, $defaults, $requirements, $tokens, $hostTokens, $requiredSchemes) = self::$declaredRoutes[$name];
at appDevUrlGenerator ->generate ('zikulaadminmodule_admin_adminpanel', array(), true) 
in vendor/jms/i18n-routing-bundle/JMS/I18nRoutingBundle/Router/I18nRouter.php at line 149   +
at I18nRouter ->generate ('zikulaadminmodule_admin_adminpanel', array(), true) 
in lib/Zikula/Bundle/CoreInstallerBundle/Stage/Upgrade/CompleteStage.php at line 57   +
at CompleteStage ->getResponse (object(Zikula_Request_Http)) 
in lib/Zikula/Bundle/CoreInstallerBundle/Controller/UpgraderController.php at line 60   +
at UpgraderController ->upgradeAction (object(Zikula_Request_Http), 'null')
at call_user_func_array (array(object(UpgraderController), 'upgradeAction'), array(object(Zikula_Request_Http), 'null')) 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php at line 145   +
at HttpKernel ->handleRaw (object(Zikula_Request_Http), '1') 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php at line 66   +
at HttpKernel ->handle (object(Zikula_Request_Http), '1', true) 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/DependencyInjection/ContainerAwareHttpKernel.php at line 64   +
at ContainerAwareHttpKernel ->handle (object(Zikula_Request_Http), '1', true) 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php at line 185   +
at Kernel ->handle (object(Zikula_Request_Http)) 
in index.php at line 25   +
'''

**Comment by craigh** [comment]
that is at the end. the upgrade should have completed successfully.

**Comment by Kaik** [comment]
This are first results I will do some more tests and will let you know.

**Comment by Portugao** [comment]
After make app/.... writable i get the button zo install. If i click on 'instaalation instructiosn' i get a white display with:

```
The file you requested (INSTALL-1.4.0.md) could not be found.
```

**Comment by Portugao** [comment]
Have to say, try to install on web.

**Comment by craigh** [comment]
> After make app/.... writable i get the button to install. If i click on 'installation instruction' i get a white display with:

This is expected because the cods aren't copied to the directory on the CI build yet.

> Have to say, try to install on web.
Try to install in subdomain.

huh? :lollipop:

**Comment by Portugao** [comment]
Ok.
Then if i start to install i get:
```
Key "config_personal_config_php" for array with keys "app/cache/, userdata/, app/config/, app/config/dynamic" does not exist in ZikulaCoreInstallerBundle:Install:requirements.html.twig at line 80 
```

**Comment by Kaik** [comment]
Ok I have made small mistake above output is what chrome says about cli upgrade which failed with this
```
back to desktop ;)
```

**Comment by craigh** [comment]
> output is what chrome says about cli upgrade which failed with this

add the `-vvv` after the command to get full trace

**Comment by Kaik** [comment]
Now web upgrade is behaving very strange.
cache cleared, both personal config and custom_parameters are present and filled with data and 
```

Unable to generate a URL for the named route "zikulaadminmodule_admin_adminpanel" as such route does not exist.

500 Internal Server Error - RouteNotFoundException

Stack Trace

in app/cache/dev/appDevUrlGenerator.php at line 53   -
    public function generate($name, $parameters = array(), $referenceType = self::ABSOLUTE_PATH)
    {
        if (!isset(self::$declaredRoutes[$name])) {
            throw new RouteNotFoundException(sprintf('Unable to generate a URL for the named route "%s" as such route does not exist.', $name));
        }
        list($variables, $defaults, $requirements, $tokens, $hostTokens, $requiredSchemes) = self::$declaredRoutes[$name];
at appDevUrlGenerator ->generate ('zikulaadminmodule_admin_adminpanel', array(), true) 
in vendor/jms/i18n-routing-bundle/JMS/I18nRoutingBundle/Router/I18nRouter.php at line 149   +
at I18nRouter ->generate ('zikulaadminmodule_admin_adminpanel', array(), true) 
in lib/Zikula/Bundle/CoreInstallerBundle/Stage/Upgrade/CompleteStage.php at line 57   +
at CompleteStage ->getResponse (object(Zikula_Request_Http)) 
in lib/Zikula/Bundle/CoreInstallerBundle/Controller/UpgraderController.php at line 60   +
at UpgraderController ->upgradeAction (object(Zikula_Request_Http), 'null')
at call_user_func_array (array(object(UpgraderController), 'upgradeAction'), array(object(Zikula_Request_Http), 'null')) 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php at line 145   +
at HttpKernel ->handleRaw (object(Zikula_Request_Http), '1') 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php at line 66   +
at HttpKernel ->handle (object(Zikula_Request_Http), '1', true) 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/DependencyInjection/ContainerAwareHttpKernel.php at line 64   +
at ContainerAwareHttpKernel ->handle (object(Zikula_Request_Http), '1', true) 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php at line 185   +
at Kernel ->handle (object(Zikula_Request_Http)) 
in index.php at line 25   +
```

**Comment by Kaik** [comment]
configs
```
parameters:
    installed: true
    env: dev
    debug: true
    compat_layer: true
    temp_dir: '%kernel.cache_dir%/ztemp'
    datadir: userdata
    system.chmod_dir: 511
    database_driver: pdo_mysql
    database_host: localhost
    database_port: null
    database_name: mleicester
    database_user: mleicester
    database_password: 'xxxxxxx'
    database_path:     ~
    database_socket:   ~
    database_server_version: 5
    # _actual_ server version doesn't matter here. Any value compensates for Doctrine Bug DBAL-1057
    # see http://www.doctrine-project.org/jira/browse/DBAL-1057
    # the actual value is written on install

    pagerendertime:    false

    locale:            en
    secret:            ThisTokenIsNotSoSecretChangeIt
    url_secret:        ~

```
```
<?php
/**
 * Copyright Zikula Foundation 2009 - Zikula Application Framework
 *
 * This work is contributed to the Zikula Foundation under one or more
 * Contributor Agreements and licensed to You under the following license:
 *
 * @license GNU/LGPL version 3 (or at your option any later version).
 * @package Zikula
 *
 * Please see the NOTICE file distributed with this source code for further
 * information regarding copyright and licensing.
 */

global $ZConfig;
$ZConfig['System'] = array();

// ----------------------------------------------------------------------
// This is the definition for the default Zikula system database.
// It must be named 'default'
// ----------------------------------------------------------------------
$ZConfig['DBInfo']['databases']['default']['host'] = 'localhost';
$ZConfig['DBInfo']['databases']['default']['user'] = 'mleicester';
$ZConfig['DBInfo']['databases']['default']['password'] = 'xxxxx';
$ZConfig['DBInfo']['databases']['default']['dbname'] = 'mleicester';
$ZConfig['DBInfo']['databases']['default']['dbdriver'] = 'mysql';
$ZConfig['DBInfo']['databases']['default']['dbtabletype'] = 'myisam';
$ZConfig['DBInfo']['databases']['default']['charset'] = 'utf8';
$ZConfig['DBInfo']['databases']['default']['collate'] = 'utf8_unicode_ci';
// additional DB can be configured here as above external2, external3 etc...

// ----------------------------------------------------------------------
// Logging Settings
// ----------------------------------------------------------------------
$ZConfig['Log']['log.apache_uname'] = 0;          // log username to apache logs: please see documentation.  Please check you country's local law covering the logging of personally identifiable user data before enabling.

// ----------------------------------------------------------------------
// The following define some data layer settings
// ----------------------------------------------------------------------
$ZConfig['System']['Z_CONFIG_USE_OBJECT_ATTRIBUTION'] = 0;     // enable universal attribution layer, 0 to turn off
$ZConfig['System']['Z_CONFIG_USE_OBJECT_CATEGORIZATION'] = 1;  // categorization/filtering services, 0 to turn off
$ZConfig['System']['Z_CONFIG_USE_OBJECT_LOGGING'] = 0;         // object audit trail logging, 0 to turn off
$ZConfig['System']['Z_CONFIG_USE_OBJECT_META'] = 0;            // meta-data services, 0 to turn off

// ----------------------------------------------------------------------
// Database cache settings
// ----------------------------------------------------------------------
$ZConfig['System']['dbcache.enable'] = 1;             // 0 to disable, 1 to enable
$ZConfig['System']['dbcache.type'] = 'Array';         // Memcache, Apc, Array, Db, Xcache (todo: Memcached, File)

// CACHE_SERVERS valid for Memcache/d only.
// array of arrays: params according to the addServer methods at e.g.
// http://php.net/manual/memcached.addservers.php or
// http://php.net/manual/function.memcache-addserver.php
$ZConfig['System']['dbcache.servers'][] = array('host' => 'localhost', 'port' => '11211', 'weight' => 1);
$ZConfig['System']['dbcache.compression'] = true; // true/false valid for dbcache.type = Memcache/d

// For pure Doctrine Queries only. Effective only when dbcache.enable = true and dbcache.cache_result = 1
// http://docs.doctrine-project.org/projects/doctrine1/en/latest/en/manual/caching.html#result-cache
$ZConfig['System']['dbcache.cache_result'] = 0;      // 1 to enable or 0 to disable.
$ZConfig['System']['dbcache.cache_result_ttl'] = 30; // seconds, 3600 = 1 hour.

// Initialize multisites array
$ZConfig['Multisites'] = array();
$ZConfig['Multisites']['multisites.enabled'] = 0;
$ZConfig['Multisites']['protected.systemvars'] = array();

```

**Comment by Kaik** [comment]
-vvv 
```
                                                             
  [Symfony\Component\Debug\Exception\ContextErrorException]  
  Notice: Undefined offset: 0                                
                                                             


Exception trace:
 () at /home/Kaik/public_html/mojeleicester/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:115
 Symfony\Component\Debug\ErrorHandler->handleError() at /home/Kaik/public_html/mojeleicester/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:115
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->installRoutesModule() at /home/Kaik/public_html/mojeleicester/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:70
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->executeStage() at /home/Kaik/public_html/mojeleicester/src/lib/Zikula/Bundle/CoreInstallerBundle/Controller/AjaxUpgradeController.php:54
 Zikula\Bundle\CoreInstallerBundle\Controller\AjaxUpgradeController->commandLineAction() at /home/Kaik/public_html/mojeleicester/src/lib/Zikula/Bundle/CoreInstallerBundle/Command/UpgradeCommand.php:108
 Zikula\Bundle\CoreInstallerBundle\Command\UpgradeCommand->execute() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php:253
 Symfony\Component\Console\Command\Command->run() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:882
 Symfony\Component\Console\Application->doRunCommand() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:195
 Symfony\Component\Console\Application->doRun() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:96
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /home/Kaik/public_html/mojeleicester/src/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at /home/Kaik/public_html/mojeleicester/src/app/console:18

```
db has changed there is empty routes table

**Comment by craigh** [comment]
looks like the Routes module failed to load the routes ping @cmfcmf

**Comment by craigh** [comment]
@Kaik - can you mail me your db?

**Comment by craigh** [comment]
@Kaik @Portugao  - now that this PR is merged, please create tickets for issues you find in further testing. I know we didn't fix everything, but it was getting impossible to follow it all in this PR :smile:

**Pull request merged**